File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String [] num_str = new String[numbers.length];
        for(int i = 0; i < num_str.length; i++) {
        	num_str[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(num_str, new Comparator<String>() {
        	public int compare(String x, String y) {
        		return (y+x).compareTo(x+y);
        	}
        });
        //아니 이거를 생각해내라고..? 다 0일경우...
        if(num_str[0].equals("0")){
            return "0";
        }
        for(int i = 0; i < num_str.length; i++) {
        	answer += num_str[i];
        }
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    var answer = numbers.map(e => String(e))
                        .sort((a, b) => (b+a) - (a+b))
                        .join('');
    // 괄호 까먹지 말기! 그리고 배열 할당도!
    return answer[0] === "0" ? "0" : answer;
}